File: python/bottle-song/bottle_song.py
def recite(start, take=1):
    """
    Recite the lyrics to the "Ten Green Bottles" song.
    
    Args:
        start: The starting number of bottles (1-10)
        take: The number of verses to recite (default: 1)
    
    Returns:
        A list of strings representing the lyrics
    """
    # Dictionary to convert numbers to words
    number_words = {
        10: "Ten",
        9: "Nine",
        8: "Eight",
        7: "Seven",
        6: "Six",
        5: "Five",
        4: "Four",
        3: "Three",
        2: "Two",
        1: "One",
        0: "no"
    }
    
    # Function to generate a single verse
    def generate_verse(num_bottles):
        bottle_word = "bottles" if num_bottles != 1 else "bottle"
        next_bottles = num_bottles - 1
        next_bottle_word = "bottles" if next_bottles != 1 else "bottle"
        
        verse = [
            f"{number_words[num_bottles]} green {bottle_word} hanging on the wall,",
            f"{number_words[num_bottles]} green {bottle_word} hanging on the wall,",
            "And if one green bottle should accidentally fall,",
            f"There'll be {number_words[next_bottles].lower()} green {next_bottle_word} hanging on the wall."
        ]
        return verse
    
    # Generate the requested verses
    lyrics = []
    for i in range(take):
        current_bottles = start - i
        if i > 0:
            lyrics.append("")  # Add empty line between verses
        lyrics.extend(generate_verse(current_bottles))
    
    return lyrics

File: javascript/bottle-song/bottle-song.js
//
// This is only a SKELETON file for the 'Bottle Song' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const recite = (initialBottlesCount, takeDownCount) => {
  const lyrics = [];
  
  for (let i = 0; i < takeDownCount; i++) {
    const currentBottleCount = initialBottlesCount - i;
    const nextBottleCount = currentBottleCount - 1;
    
    // Add verse
    lyrics.push(bottleLine(currentBottleCount));
    lyrics.push(bottleLine(currentBottleCount));
    lyrics.push(`And if one green bottle should accidentally fall,`);
    lyrics.push(`There'll be ${bottlePhrase(nextBottleCount)} hanging on the wall.`);
    
    // Add empty line between verses (except after the last verse)
    if (i < takeDownCount - 1) {
      lyrics.push(``);
    }
  }
  
  return lyrics;
};

// Helper function to generate the bottle line
function bottleLine(count) {
  return `${capitalize(bottlePhrase(count))} hanging on the wall,`;
}

// Helper function to generate the bottle phrase
function bottlePhrase(count) {
  if (count === 0) {
    return 'no green bottles';
  } else if (count === 1) {
    return 'one green bottle';
  } else {
    return `${numberToWord(count)} green bottles`;
  }
}

// Helper function to convert numbers to words
function numberToWord(num) {
  const words = [
    'zero', 'one', 'two', 'three', 'four',
    'five', 'six', 'seven', 'eight', 'nine', 'ten'
  ];
  
  return num <= 10 ? words[num] : num.toString();
}

// Helper function to capitalize the first letter of a string
function capitalize(str) {
  return str.charAt(0).toUpperCase() + str.slice(1);
}
